stl/inc/atomic

```diff
@@ -304,63 +314,100 @@ template <class _Ty>
 struct _Atomic_padded<_Ty, false> {
     alignas(sizeof(_Ty)) mutable _Ty _Value; // align to sizeof(T); x86 stack aligns 8-byte objects on 4-byte boundaries
 };
+
+template <class _Ty>
+struct _Atomic_padded<_Ty&, false> {
+    _Ty& _Value;
+};
+
 #endif // TRANSITION, ABI
 
+template <class _Ty>
+struct _Atomic_storage_types {
+    using _TVal     = _Ty;
+    using _TStorage = _Atomic_padded<_Ty>;
+    using _TConstr  = const _Ty;
+    using _Spinlock = long;
+
+    static constexpr bool _Has_volatile = true;
+
+    static long* _Get_spinlock(long& _Spinlock) {
+        return &_Spinlock;
+    }
+};
+
+template <class _Ty>
+struct _Atomic_storage_types<_Ty&> {
+    using _TVal     = _Ty;
+    using _TStorage = _Ty&;
+    using _TConstr  = _Ty&;
+    using _Spinlock = long*;
+
+    static constexpr bool _Has_volatile = false;
+
+    static long* _Get_spinlock(long* _Spinlock) {

```

Comment:
I think this should use an SRWLOCK directly rather than trying to maintain the spinlock for `atomic_ref`.

Reply:
We would not be reusing `atomic` spinlock then, and will have `_CONSTEXPR_IF` branches in locking. Is this ok?

Reply:
I think that's OK.

Reply:
I put _Smtx_t from `<xatomic.h>` (call exclusive functions only)